src/domain/community/organization/organization.service.ts

```diff
@@ -307,9 +308,28 @@ export class OrganizationService {
     return organization;
   }
 
-  async getOrganizations(): Promise<IOrganization[]> {
-    const organizations = await this.organizationRepository.find();
-    return organizations || [];
+  async getOrganizations(
+    limit?: number,
+    randomSelection = false
+  ): Promise<IOrganization[]> {
+    const organizations: IOrganization[] =
+      await this.organizationRepository.find();
+    if (!organizations) return [];
+    if (!limit || organizations.length < limit) return organizations;
```

Comment:
you are not taking in account the `randomSelection` param

Reply:
addressed in 9a57bcd8860c2ff4031bed6d103313c4b6c9ba8f

Reply:
Note: shuffle paramter is only applied in the case where a limit is set.